### PR TITLE
Improve container admission exception lifecycle gates

### LINKED ISSUE #676: [REVIEW] container-security: add admission exception owner and TTL checks
# [REVIEW] container-security: add admission exception owner and TTL checks

## Skill Being Reviewed
**Skill name:** `container-security`
**Skill path:** `skills/cloud/container-security/`

## False Positive Analysis

The skill can overstate risk when a privileged or unsigned workload runs under a documented temporary exception. Some production incidents require a narrow break-glass admission bypass, but it should have owner, TTL, namespace scope, and compensating controls.

Benign example:

```text
namespace: incident-response
exception: allow privileged debug pod
ttl: 4 hours
owner: incident commander
controls: node taint, audit log, post-incident deletion
```

This should be treated as a controlled temporary exception, not the same as a permanent policy bypass.

## Coverage Gaps

The container review covers image and Kubernetes hardening, but it does not require lifecycle evidence for admission-control exceptions.

Missed variant 1: a namespace label disables Pod Security Admission or Kyverno/Gatekeeper checks permanently.

Missed variant 2: an image-signature exception allows unsigned images from a broad registry path instead of one digest.

Missed variant 3: a privileged workload exception has no expiry, owner, or follow-up ticket, so a temporary bypass becomes a standing control gap.

## Edge Cases

- System namespaces often need special rules; they still need documented scope and review cadence.
- Some CNI, CSI, and monitoring agents legitimately require host access.
- CI test clusters may allow wider exceptions, but production controls should be evaluated separately.

## Remediation Quality

- Add exception evidence fields: policy name, namespace, workload, owner, reason, expiry, and compensating controls.
- Require exceptions to bind to image digest or exact workload identity where possible.
- Require stale-exception detection and evidence of post-expiry cleanup.
- Add false-positive examples for emergency debug sessions and system daemonsets.

## Comparison to Other Tools

Kyverno, Gatekeeper, and admission-controller logs can show what was allowed, but they do not prove exceptions are temporary and owned. The skill should make that lifecycle explicit.

## Overall Assessment

Strong container baseline. Main gap: policy exceptions are a common failure mode and need owner/TTL evidence to distinguish controlled risk from silent bypass.

## Bounty Info

- [x] I have read and agree to the `CONTRIBUTING.md` bounty terms.
- Preferred payment method: PayPal `samik4184@gmail.com`

### PR DESCRIPTION
**Pull request by tzh476**
Closes #676.

## Summary
- Add an admission exception lifecycle gate for Pod Security Admission, Kyverno/Gatekeeper, image policy, and admission-control waivers.
- Require owner, reason, ticket/change reference, expiry/TTL, scope minimization, compensating controls, audit/report evidence, and stale-exception cleanup.
- Extend the detailed CIS checklist with vulnerable and controlled exception patterns plus review checks.
- Add calibration fixtures for permanent privileged namespace exceptions, broad unsigned-image exceptions, and controlled incident-response debug exceptions.
- Sync container-security discovery metadata with the new admission-control coverage.

## Validation
- git diff --cached --check
- container-security metadata sync check
- admission exception coverage keyword check
- markdown fence-balance check
- official source URL checks for Kubernetes PSA, ValidatingAdmissionPolicy, Kyverno PolicyException, and Gatekeeper namespace exemptions
- prompt-injection/old-email scan

## Bounty
Improver tier; this is a moderate false-positive reduction and edge-case coverage improvement. Requested bounty: $100.

### CODE REVIEW

File: skills/cloud/container-security/tests/admission-exception-lifecycle.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+apiVersion: kyverno.io/v2
+kind: PolicyException
+metadata:
+  name: ir-debug-hostaccess-20260605
+  namespace: policy-exceptions
+  annotations:
+    security.example.com/owner: incident-commander
+    security.example.com/ticket: INC-2026-0605
+    security.example.com/reason: "4h privileged debug pod on tainted IR node"
+    security.example.com/expires-at: "2026-06-05T08:00:00Z"
```
[the remark this answers is not in the document]
Comment (tzh476) [reply skills/cloud/container-security/tests/admission-exception-lifecycle.md]: Thanks, good catch. I replaced the fixed `2026-06-05T08:00:00Z` value with a relative `<incident-start + 4h RFC3339>` placeholder in both the checklist example and calibration fixture so the benign case does not become stale as the repository ages. Commit: 530f4e3.

File: skills/cloud/container-security/tests/admission-exception-lifecycle.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    - policyName: require-signed-images
+      ruleNames:
+        - "*"
+  match:
+    any:
+      - resources:
+          namespaces:
+            - "*"
+          images:
+            - "registry.example.com/tools/*:latest"
```
[the remark this answers is not in the document]
Comment (tzh476) [reply skills/cloud/container-security/tests/admission-exception-lifecycle.md]: Thanks, good catch. I updated the broad unsigned-image exception sample to use the current `policies.kyverno.io/v1beta1` PolicyException shape with `policyRefs`, top-level `images`, and `matchConditions`, and removed the incorrect `match.any[].resources.images` form. Commit: 530f4e3.
